tests/dynamic_honey_badger.rs

```diff
@@ -99,11 +102,65 @@ where
         if !input_add && network.nodes.values().all(has_remove) {
             network.input_all(Input::Change(Change::NodeChange(pub_keys_add.clone())));
             input_add = true;
+        } else if !rejoined_node0 {
+            if let Some(join_plan) = network
+                .nodes
+                .values()
+                .flat_map(|node| node.outputs())
+                .filter_map(|batch| match batch.change() {
+                    ChangeState::Complete(Change::NodeChange(pub_keys))
```

Comment:
Shouldn't we add the joining node already on `InProgress`?

Reply:
Not sure. It's still failing...

Reply:
Fixed it. There was an issue with a message from node 0 not being sent. Also I updated the batch verification function to allow the test pass when node 0 is missing some batches in the middle.